DPAT 2204 Guide for using open source shiny server (#382)
moj-analytical-services/user-guidance: source/documentation/apps/rshiny-app.md

Majority of the shiny apps hosted on CP are using this version,  the current tag for this docker image is 
`593291632749.dkr.ecr.eu-west-1.amazonaws.com/rshiny:r4.1.3-shiny0.0.6`

### Open source shiny-server

We also provide the solution of using [open source one](https://github.com/rstudio/shiny-server) with a few minor tweaks to support `USER_EMAIL`and `COOKIE` headers.  The base docker image is defined [here](https://github.com/ministryofjustice/data-platform/blob/main/containers/rshiny-open-source-base/Dockerfile). The version of open source shiny server is defined by `SHINY_SERVER_VERSION`, current is `1.5.20.1002`.

It offers more features than the DPAT own one and supports 
- Better reconnection behaviour:
  - Reconnect and load existing session rather than creating a new session automatically 
  - If failed to reconnect eventually, the window popped up for manual-reconnection will reload the app rather than retriggering same reconneciton behaviour
- Better session management:
  - The session will be closed when session is idle for certain period of time.

> which can 

Reviewer: ```suggestion
This behaviour can result in:
```

> - Keep the session data(reactive values) even reconnection happens

Reviewer: ```suggestion
- Session data (reactive values) is retained even after a reconnection happens
```

> - Release the resources e.g., memory linked to the session whhich avoid potential memory leaking

Reviewer: ```suggestion
- Release resources e.g. memory linked to the session which avoids potential memory leaking
```

> It also provides more [configuration options (non-Pro ones)](https://docs.posit.co/shiny-server/)

Reviewer: ```suggestion
It also provides more [configuration options as outlined here](https://docs.posit.co/shiny-server/). Note: options marked as "pro" are not available
```

> #### Planning to use open source shiny server 

Reviewer: ```suggestion
#### Instructions for using the open source shiny server image
```

#   the one in the example below has been provided in base image.
# RUN apt-get update \
#   && apt-get install -y --no-install-recommends \
#     libglpk-dev

# use renv for packages
ADD renv.lock renv.lock

# Install R packages
RUN R -e "install.packages('renv'); renv::restore()"

# ** Optional step: only if the app requires python packages
# Make sure reticulate uses the system Python
# ENV RETICULATE_PYTHON="/usr/bin/python3"
# ensure requirements.txt exists (created automatically when making a venv in renv)
# COPY requirements.txt requirements.txt
# RUN python3 -m pip install -r requirements.txt

# Add shiny app code
ADD . .

USER 998
```

If you switch to open source server from the DPAT own one, the key changes you need to make are 
- Switching to use new base docker image 
```
FROM ghcr.io/ministryofjustice/data-platform-rshiny-open-source-base:1.0.3
```
- Removing the followingr reduncant parts from Dockerfile 
```
ENV PATH="/opt/shiny-server/bin:/opt/shiny-server/ext/node/bin:${PATH}"
ENV SHINY_APP=/srv/shiny-server
ENV NODE_ENV=production

CMD analytics-platform-shiny-server
EXPOSE 9999
```

##### GitHub work flows

> Assume the app uses the GitHub flow from DPAT, the following parameters for helm installtion are required

Reviewer: ```suggestion
Assuming the app uses the GitHub workflows from AP, the following parameters for helm installation are required in both the `build-push-deploy-dev.yml` and `build-push-deploy-prod.yml` github action workflow files:
```
